src/library/Cardano/Fee.hs

```diff
@@ -128,21 +126,47 @@ newtype FeeEstimator i o = FeeEstimator
 data FeeOptions i o = FeeOptions
     { feeEstimator
         :: FeeEstimator i o
+        -- ^ Estimate fees based on selected inputs and requested outputs.
+
     , dustThreshold
         :: DustThreshold
+        -- ^ Change addresses below the given threshold will be evicted
+        -- from the created transaction. Setting 'dustThreshold' to 0
+        -- removes output equal to 0
+
+    , onDanglingChange
+        :: OnDanglingChange
+        -- ^ What do to when we encouter a dangling change output.
+        -- See 'OnDanglingChange'
     } deriving Generic
 
+-- | We call 'dangling' a change output that would be too expensive to add. This
+-- can happen when a change output is small, but adding it generate more fees
+-- than not having it.
+--
+-- In case where nodes accept slightly unbalanced transactions, we may choose to
+-- save money and keep the transaction slightly unbalanced. In case where node
+-- demands exactly balanced transaction, we have no choice but to add the extra
+-- dangling change output and pay for the extra cost induced.
```

Comment:
> Can it be stated in the form of a property involving a CoinSelection and estimateFee?

It is already!

```
            -- If fees are null and we do not require balanced selections, then
            -- the selection must leave __at least__ the fee amount.
            SaveMoney | remainder == Fee C.zero ->
                (Fee <$> delta sel') >= Just fee
```

Reply:
@jonathanknowles wrote:
> > Can it be stated in the form of a property involving a CoinSelection and estimateFee?
@KtorZ replied:
> It is already!

I was thinking more along the lines of adding a simple statement to the **Haddock documentation**, such as this one:

```hs
-- A selection is /balanced/ if and only if it satisfies the following
-- property:
--
-- >>> sumInputs c == sumOutputs c + sumChange c + estimateFee c
```

If a library user encounters the term "balanced" while reading through the Haddock documentation and doesn't understand what it means, it would be great if there were a definition available (without requiring them to go hunting through the source code!).

Reply:
Arguably, if someone doesn't know what a "balanced transaction" is, they shouldn't be using this library in the first place :wink:

Reply:
> Arguably, if someone doesn't know what a "balanced transaction" is, they shouldn't be using this library in the first place wink

Well, I'm not so convinced about this one.

For example, the following searches don't turn up many useful results (with the search engines I tried):

* `balanced transaction` (produces a lot of junk unrelated to Blockchain)
* `"balanced transaction" blockchain` (produces a lot of definitions related to Elliptic Curves)
* `"balanced transaction" UTxO` (we're getting closer, but there are lots of articles that use it without defining it).

The Cardano documentation for "transaction" doesn't even mention the term "balanced": https://docs.cardano.org/cardano/transactions/

I suspect that our users may come with different levels of experience, and that it won't hurt to state something concretely.